pal03377/Athena-New
# General improvements for `module_programming_llm`

### Motivation and Context



The `module_programming_llm` has been neglected a bit and I need to make it ready for presentation. This PR aims at making everything ready.

### Description


- Huge refactor of the module
- Use the model config from `module_text_llm` (just copy paste), also copied some other helpers 1 to 1
    * `module_programming_llm/helpers/llm_utils.py` copied
    * `module_programming_llm/helpers/models/*` copied
    * Env files are also copied
- Support function calling for OpenAI chat models and use structured output parsing similar to `module_text_llm`
- Remove programming language file extension limitation (LLMs can handle most file types anyways)
- Rework splitting of problem statement and grading instructions (it is now in the feedback generation pipeline, instead with receive submissions)
- General improvements to improve reliability and feedback quality

### Steps for Testing

1. Use `.env` file from `module_text_llm` in `module_programming_llm`
2. Start module and playground as always
3. Check out the new module's config if everything looks good in the playground
4. Generate feedback suggestions using `module_programming_llm` successfully (you don't have to test the Replicate models)

### Screenshots
